x/stakeibc/keeper/icacallbacks_delegate_test.go

```diff
@@ -0,0 +1,270 @@
+
+	// Confirm deposit record has been removed
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx())
+	s.Require().Len(records, 1, "number of deposit records")
+	record := records[0]
+	s.Require().Equal(recordtypes.DepositRecord_STAKE, record.Status, "deposit record status should be timeout")
+}
+
+func (s *KeeperTestSuite) TestDelegateCallback_HostNotFound() {
+	tc := s.SetupDelegateCallback()
+	validArgs := tc.validArgs
+	s.App.StakeibcKeeper.RemoveHostZone(s.Ctx(), chainId)
+	err := stakeibckeeper.DelegateCallback(s.App.StakeibcKeeper, s.Ctx(), validArgs.packet, validArgs.ack, validArgs.args)
+	s.Require().EqualError(err, "host zone not found GAIA: invalid request")
+
+	// Confirm deposit record has been removed
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx())
+	s.Require().Len(records, 1, "number of deposit records")
+	record := records[0]
+	s.Require().Equal(recordtypes.DepositRecord_STAKE, record.Status, "deposit record status should be timeout")
+}
+
+func (s *KeeperTestSuite) TestDelegateCallback_BigAmount() {
+	tc := s.SetupDelegateCallback()
+	validArgs := tc.validArgs
+	badSplitDelegation := types.SplitDelegation{
+		Validator: 		"address",
+		Amount:			math.MaxUint64,
+	}
+	callbackArgs := types.DelegateCallback{
+		HostZoneId:				chainId,
+		DepositRecordId:		1,
+		SplitDelegations:		[]*types.SplitDelegation{&badSplitDelegation},
+	}
+	args, err := s.App.StakeibcKeeper.MarshalDelegateCallbackArgs(s.Ctx(), callbackArgs)
+	s.Require().NoError(err)
+	err = stakeibckeeper.DelegateCallback(s.App.StakeibcKeeper, s.Ctx(), validArgs.packet, validArgs.ack, args)
+	s.Require().ErrorContains(err, "overflow: unable to cast")
+
+	// Confirm stakedBal has not increased
+	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), chainId)
+	s.Require().True(found)
+	s.Require().Equal(int64(tc.initialState.stakedBal), int64(hostZone.StakedBal), "stakedBal should not have increased")
+
+	// Confirm deposit record has been removed
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx())
+	s.Require().Len(records, 1, "number of deposit records")
+	record := records[0]
+	s.Require().Equal(recordtypes.DepositRecord_STAKE, record.Status, "deposit record status should be timeout")
+}
+
+func (s *KeeperTestSuite) TestDelegateCallback_MissingValidator() {
+	tc := s.SetupDelegateCallback()
+	validArgs := tc.validArgs
+	badSplitDelegation := types.SplitDelegation{
+		Validator: 		"address_dne",
+		Amount:			1234,
+	}
+	callbackArgs := types.DelegateCallback{
+		HostZoneId:				chainId,
+		DepositRecordId:		1,
+		SplitDelegations:		[]*types.SplitDelegation{&badSplitDelegation},
+	}
+	args, err := s.App.StakeibcKeeper.MarshalDelegateCallbackArgs(s.Ctx(), callbackArgs)
+	s.Require().NoError(err)
+	err = stakeibckeeper.DelegateCallback(s.App.StakeibcKeeper, s.Ctx(), validArgs.packet, validArgs.ack, args)
+	s.Require().ErrorContains(err, "Failed to add delegation to validator")
```

Comment:
I think the `ErrorContains` here is clean. But it's also kind of nice to see the full error message in these tests (since we have a mix of a few different flavors of error wrapping). Wdyt?

Reply:
done